source/sigv4.c

```diff
@@ -1239,14 +1259,21 @@ static SigV4Status_t generateCredentialScope( const SigV4Parameters_t * pSigV4Pa
             }
             else if( isAllowedChar( *pUriLoc, encodeSlash ) )
             {
-                *pBufLoc = *pUriLoc;
-                ++pBufLoc;
-                ++bytesConsumed;
+                if( bytesConsumed < bufferLen )
+                {
+                    *pBufLoc = *pUriLoc;
+                    ++pBufLoc;
+                    ++bytesConsumed;
+                }
+                else
+                {
+                    returnStatus = SigV4InsufficientMemory;
+                    LOG_INSUFFICIENT_MEMORY_ERROR( "encode the URI", bytesConsumed + 1U - bufferLen );
+                }
             }
             else
             {
-                if( ( bytesConsumed > ( SIZE_MAX - URI_ENCODED_SPECIAL_CHAR_SIZE ) ) ||
-                    ( ( bytesConsumed + URI_ENCODED_SPECIAL_CHAR_SIZE ) > bufferLen ) )
+                if( ( bufferLen - bytesConsumed ) < URI_ENCODED_SPECIAL_CHAR_SIZE )
```

Comment:
We should probably check `bufferLen` > `bytesConsumed`

Reply:
No, since `bytesConsumed` is initialized to 0 and `bufferLen` is initialized to `*canonicalURILen` which is >= 0 as these are all `size_t`s. In every branch there is a check that there is enough space in the buffer before writing any bytes, so it's never possible for `bytesConsumed > bufferLen`

Reply:
Okay, I suppose CBMC will just catch if there are any misses